pytorch_lightning/core/datamodule.py

```diff
@@ -260,13 +274,41 @@ def has_setup_predict(self) -> bool:
         """
         return self._has_setup_predict
 
-    @abstractmethod
-    def prepare_data(self, *args, **kwargs):
-        pass
+    @property
+    def has_teardown_fit(self) -> bool:
+        """Return bool letting you know if ``datamodule.teardown(stage='fit')`` has been called or not.
 
-    @abstractmethod
-    def setup(self, stage: Optional[str] = None):
-        pass
```

Comment:
Should we keep these two abstract methods, now that we already have them by inheriting from `DataHooks`?